### ISSUE TITLE
ValidatorAddress in CommitSig

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#5023]
There's been much discussion around the presence of the ValidatorAddress in CommitSig. Most recently in #4835.

Ultimately it is redundant and takes up unnecessary space in blocks, eg. for 100 validators, it's an extra 2kB per block, which does add up.

In [ADR-025](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-025-commit.md) where the CommitSig were introduced to replace other redundancies in the Commit, it was decided to keep the ValidatorAddress for now. Having it is a major convenience in a number of ways, since it avoids the need to pass validator sets through various APIs and still know who signed which commit. 

This is especially important in the light client and evidence handling. Light client bisection requires checking whether the signers for a given commit constitute +1/3 of an older validator set. All the code currently uses the ValidatorAddress in the commit. Without the ValidatorAddress, they'd have to first get the validator address from the current validator set, and then check if that validator existed in the older validator set. It was suggested that having the address in the commit sig can save on bandwidth, since the client doesn't need to fetch the validator set right away, but in practice the clients seem to be written to fetch the val set anyways, so the savings are only theoretical. In the long term, the savings from not having the ValidatorAddress in the commitsig will probably be significant and are ultimately worth it.

There's also our evidence handling logic, which uses the ValidatorAddress to check if the signature is from a legit validator for the given height. Without it, the light clients will need to additionally submit the validator set along with the signed header so full nodes can check this sort of thing.

In short, there's a non-trivial amount of work that needs to be done across the codebase to get the ValidatorAddress out. I propose that work be done iteratively (like was done with CommitSig in the first place), working across the APIs to reduce the dependency on the ValidatorAddress by passing in validator sets where necessary, until none of the code depends on accessing CommitSig.ValidatorAddress. In general this will be breaking on the APIs, but not on the block protocol. Once this is all done, it will then be trivial to just delete the ValidatorAddress in a final protocol breaking change (eg. to be done with the BFT proposer time).

**Comment by ebuchman** [comment]
Related/impacted by this (please update this list as necessary to track things):
- #2561 - Check Evidence By Validator Address
